include/type_safe/detail/aligned_union.hpp

```diff
@@ -41,11 +42,11 @@ namespace detail
         static constexpr auto size_value      = detail::max(sizeof(Types)...);
         static constexpr auto alignment_value = detail::max(alignof(Types)...);
 
-        using type = typename std::aligned_storage<size_value, alignment_value>::type;
+        alignas(alignment_value) unsigned char storage_[size_value];
     };
 
     template <typename... Types>
-    using aligned_union_t = typename aligned_union<Types...>::type;
+    using aligned_union_t = aligned_union<Types...>;
```

Comment:
That's not what I meant.

Please update https://github.com/foonathan/type_safe/blob/8d4a501414eda0015e49dc807588c27c56c28821/include/type_safe/tagged_union.hpp#L259, to use `alignas(alignment_value) storage_[size_value]` directly, then drop `aligned_union_t` and `aligned_union::storage_`.

Reply:
I am very confused I'm not sure what you mean by "directly". Do you mean get rid of the aligned_union althogether and then use alignas anywhere in the code that used to use aligned_union? If I use `alignas(alignment_value) storage_[size_value]` then alignment_value and size_value must live somewhere. Do you mean use `alignas([alignment_value](detail::max(sizeof(Types)...))) storage_[[size_value](detail::max(alignof(Types)...))]` elsewhere in the code?

Reply:
For example tagged_union ha the code
```
    using storage_t = detail::aligned_union_t<Types...>;
    storage_t storage_;
```
Do you mean directly us alignas here?

Reply:
> Do you mean directly us alignas here?

Yes, use:

````cpp
alignas(detail::aligned_union<Types...>::alignment_value) unsigned char storage_[detail::aligned_union<Types...>::size_value];
```

If that's the only use of `detail::aligned_union`, you can also get rid of it and do the size/alignment computation directly inside `tagged_union`.

Reply:
So I tried do the aligned array direclty in the tagged_union like
```
    alignas(detail::aligned_union<Types...>::alignment_value) unsigned char storage_[detail::aligned_union<Types...>::size_value];
```
I then get compile errors
`tagged_union.hpp(419,32): error C3863: array type 'unsigned char [12]' is not assignable`

Reply:
Okay, arrays are annoying, but unfortunately we have to use them - structs cannot provide storage according to the standard. I think you can replace that line (and any other offending assignments) by memcpy.